cmd/virt-launcher/virt-launcher.go

```diff
@@ -601,3 +611,12 @@ func RemoveContents(dir string) error {
 	}
 	return nil
 }
+
+func istioProxyPresent() bool {
+	cmd := "ss | grep istio/proxy"
+	err := exec.Command("/bin/sh", "-c", cmd).Run()
+	if err != nil {
+		return false
+	}
+	return true
```

Comment:
is there a golang native way of detecting this without using an external tool?

Reply:
[Updated](https://github.com/kubevirt/kubevirt/compare/3146e3ab1a2bd5e8cbb463c9a79c052e8e0a98a3..a30725bfddcef787f19fa558cdbf951b13574c92). To avoid using `ss` I send an http GET request to a localhost port that enovoy uses for health probes.
To indentify Envoy, I check if the response header contains `server: enovy`.